src/main/antlr/SHRDataElementParser.g4

```diff
@@ -13,16 +13,16 @@ pathDefs:           pathDef+;
 pathDef:            KW_PATH ALL_CAPS EQUAL URL;
 
 vocabularyDefs:     vocabularyDef+;
-vocabularyDef:      KW_VOCABULARY ALL_CAPS EQUAL (URL | URN_OID | URN);
+vocabularyDef:      KW_VOCABULARY ALL_CAPS EQUAL (URL | URN_OID | URN); // (KW_VOCAB_VERSION WHOLE_NUMBER DOT WHOLE_NUMBER)?;
 
 dataDefs:           dataDef*;
 dataDef:            elementDef | entryDef;
 
 elementDef:         elementHeader elementProps? values;
-elementHeader:      KW_ELEMENT simpleName;
+elementHeader:      KW_ABSTRACT? KW_ELEMENT simpleName;
 
 entryDef:           entryHeader elementProps? values;
-entryHeader:        KW_ENTRY_ELEMENT simpleName;
+entryHeader:        KW_ABSTRACT? KW_ENTRY_ELEMENT simpleName;
```

Comment:
Does it make sense to have an abstract entry?  My guess is that it probably only makes sense on `elementHeader` since you couldn't instantiate an abstract entry anyway.  I think entries should always be instantiable.